subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Implimentation for Bridging Table state of Subnet Bridge feature
+ * in Bluetooth Mesh Protocol v1.1 specification
+ */
+
+#include <zephyr/kernel.h>
+#include <string.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/mesh.h>
+
+#include "mesh.h"
+#include "net.h"
+#include "subnet.h"
+#include "settings.h"
+#include "brg_cfg.h"
+
+#define LOG_LEVEL CONFIG_BT_MESH_NET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_mesh_brg_cfg);
+
+/* Bridging table state. Each item is a slist node. */
+static struct brg_tbl_row brg_tbl[CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX];
+static uint32_t brg_tbl_row_cnt;
+static bool brg_enabled;
+
+static void brg_tbl_compact(void)
+{
+	int j = 0;
+
+	for (int k = 0; k < brg_tbl_row_cnt; k++) {
+		if (brg_tbl[k].direction != 0) {
+			brg_tbl[j] = brg_tbl[k];
+			j++;
+		}
+	}
+	memset(&brg_tbl[j], 0, sizeof(brg_tbl[j]));
+	brg_tbl_row_cnt--;
+}
+
+static int brgen_set(const char *name, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	int err;
+
+	if (len_rd == 0) {
+		brg_enabled = 0;
+		LOG_DBG("Cleared brgen");
+		return 0;
+	}
+
+	err = bt_mesh_settings_set(read_cb, cb_arg, &brg_enabled, sizeof(brg_enabled));
+	if (err) {
+		LOG_ERR("Failed to set brgen");
```

Comment:
The comment should not use "set brgen", but rather use "set subnet bridge state". Same comment in the line 60 and 70.